### ISSUE TITLE
Naming issues with custom Query root type

### ISSUE THREAD

**Issue by eliasdarruda** [zhouzi/graphql-codegen-factories#75]
I think there is a issue when the __typename doesnt follow the "Query" or "Mutation" pattern

Testing the following query (querying through hasura)

```graphql
query Schedules {
  schedules {
    id
    start_hour
  }
}
```

`typescript-operations` generates for me:

```TS
export type SchedulesQuery = {
  __typename?: 'query_root',
  schedules: Array<{ __typename?: 'schedules', id: number, start_hour: string }>
}
```

## What `graphql-codegen-factories/operations` generates for me is:

```TS
export function createSchedulesquery_RootMock(props: Partial<Schedulesquery_Root> = {}): Schedulesquery_Root {
  return {
    __typename: "query_root",
    schedules: [],
    ...props,
  };
}
```

## The expected output here should be:

```TS
export function createSchedulesQueryMock(props: Partial<SchedulesQuery> = {}): SchedulesQuery {
  return {
    __typename: "query_root",
    schedules: [],
    ...props,
  };
}
```

-------

Here is what I'm using for my `codegen.ts` file:

```TS
import { CodegenConfig } from '@graphql-codegen/cli'

const config: CodegenConfig = {
  schema: 'http://.....',
  documents: ['src/**/*.ts'],
  ignoreNoDocuments: true, // for better experience with the watcher
  generates: {
    './src/_generated_gql/': {
      preset: 'client',
      plugins: [
        'graphql-codegen-factories/schema',
        'graphql-codegen-factories/operations',
      ],
    },
  },
}

export default config
```

**Comment by zhouzi** [comment]
Thanks for reporting! Could you share the configuration you are using with `typescript-operations`, please?

**Comment by eliasdarruda** [comment]
> Thanks for reporting! Could you share the configuration you are using with `typescript-operations`, please?

Its implicitly included in the [client preset](https://the-guild.dev/graphql/codegen/plugins/presets/preset-client)

**Comment by zhouzi** [comment]
I was able to reproduce this issue with:

```graphql
query CustomQuery {
  ...
}

schema {
  query: CustomQuery
}
```

I'm looking into it and will let you know once the fix is released 👌

**Comment by zhouzi** [comment]
@eliasdarruda this is now fixed as of v1.2.1. Feel free to give it a go and let me know if you're facing any other issues when using it with the client presets and hasura 😉

**Comment by eliasdarruda** [comment]
Thanks a lot! it worked
